### PR TITLE
vm/scanners

### PR DESCRIPTION
**Pull request by harshshah-tenable**
# Updated following actions for Scanners API:
- search()
- get_permissions()
- edit_permissions()

Other structure and schema related changes are also covered in this PR

**ToDo:** 
- Validate search method and tests once search WUG PR is merged
- Validate get_permissions() and edit_permissions() 
- allowed_scanners() depends policies #39 and editors #51 PRs  update/implement once dependent PR's are merged in develop

### CODE REVIEW

File: tenable/io/v3/vm/scanners/api.py [hunk 1]
```diff
@@ -159,9 +162,9 @@ def details(self, id: UUID) -> Dict:
 
         Examples:
             >>> scanner = tio.v3.vm.scanners.details(1)
-            >>> pprint(scanner)
+            ... pprint(scanner)
```
Comment (ngaglani-tenable) [inline tenable/io/v3/vm/scanners/api.py]:
```suggestion
            >>> pprint(scanner)
```

File: tests/io/v3/vm/scanners/test_scanners_api.py [hunk 2]
```diff
@@ -216,12 +245,94 @@ def test_scanners_get_scans(api):
...
+
+    iterator = api.v3.vm.scanners.search(
+        fields=fields, limit=200, sort=sort
+    )
+    assert isinstance(iterator, SearchIterator)
+
+    event_list = []
+    for event in iterator:
+        event_list.append(event)
+    assert len(event_list) == api_response['pagination']['total']
```
Comment (ngaglani-tenable) [inline tests/io/v3/vm/scanners/test_scanners_api.py]:
```suggestion
    scnners_list = []
    for scanner in iterator:
        scnners_list.append(scanner)
    assert len(scnners_list) == api_response['pagination']['total']
```

File: tenable/io/v3/vm/scanners/api.py [hunk 3]
```diff
@@ -12,19 +12,23 @@
...
 from typing_extensions import Literal
 
 from tenable.io.v3.base.endpoints.explore import ExploreBaseEndpoint
-from tenable.io.v3.vm.scanners.schema import ScannerEditSchema
+from tenable.io.v3.base.iterators.explore_iterator import (CSVChunkIterator,
+                                                           SearchIterator)
+from tenable.io.v3.vm.scanners.schema import ScannerSchema
 
 
 class ScannersAPI(ExploreBaseEndpoint):
```
Comment (ngaglani-tenable) [inline tenable/io/v3/vm/scanners/api.py:31]: Class docstring is missing.

File: tests/io/v3/vm/scanners/test_scanners_api.py [hunk 4]
```diff
@@ -216,27 +245,90 @@ def test_scanners_get_scans(api):
...
+        f'{SCANNER_BASE_URL}/search',
+        match=[matchers.json_params_matcher(payload)],
+        json=api_response
+    )
 
+    iterator = api.v3.vm.scanners.search(
+        fields=fields, limit=200, sort=sort
+    )
+    assert isinstance(iterator, SearchIterator)
+    assert len(list(iterator)) == api_response['pagination']['total']
```
Comment (alpesh-te) [inline tests/io/v3/vm/scanners/test_scanners_api.py:321]: Get rid of loop.

**Review by alpesh-te — CHANGES_REQUESTED** [review]
Minor fixes!

File: tests/io/v3/vm/scanners/test_scanners_api.py (hunk 4 above)
Comment (alpesh-te) [reply tests/io/v3/vm/scanners/test_scanners_api.py:321]: It is already fixed

**Review by alpesh-te — APPROVED** [review]

**Pull request merged**